pkg/kubelet/volume/nfs_mount/nfs_mount.go

```diff
@@ -0,0 +1,169 @@
+	plugin.host = host
+}
+
+func (plugin *nfsMountPlugin) Name() string {
+	return nfsMountPluginName
+}
+
+func (plugin *nfsMountPlugin) CanSupport(spec *api.Volume) bool {
+	if spec.Source.NFSMount != nil {
+		return true
+	}
+	return false
+}
+
+func (plugin *nfsMountPlugin) NewBuilder(spec *api.Volume, podUID types.UID) (volume.Builder, error) {
+	return plugin.newBuilderInternal(spec, podUID, mount.New())
+}
+
+func (plugin *nfsMountPlugin) newBuilderInternal(spec *api.Volume, podUID types.UID, mounter mount.Interface) (volume.Builder, error) {
+
+	return &nfsMount{
+		volName:      spec.Name,
+		server:       spec.Source.NFSMount.Server,
+		exportPath:   spec.Source.NFSMount.SourcePath,
+		mountOptions: spec.Source.NFSMount.MountOptions,
+		podUID:       podUID,
+		mounter:      mounter,
+		plugin:       plugin,
+	}, nil
+}
+
+func (plugin *nfsMountPlugin) NewCleaner(volName string, podUID types.UID) (volume.Cleaner, error) {
+	return plugin.newCleanerInternal(volName, podUID, mount.New())
+}
+
+func (plugin *nfsMountPlugin) newCleanerInternal(volName string, podUID types.UID, mounter mount.Interface) (volume.Cleaner, error) {
+	return &nfsMount{
+		volName:      "",
+		server:       "",
+		exportPath:   "",
+		mountOptions: "",
+		podUID:       podUID,
+		mounter:      mounter,
+		plugin:       plugin,
+	}, nil
+}
+
+// NFSMount volumes represent a bare host file or directory mount of an NFS export.
+// The direct at the specified path will be directly exposed to the container.
+type nfsMount struct {
+	volName      string
+	podUID       types.UID
+	server       string
+	exportPath   string
+	mountOptions string
+	mounter      mount.Interface
+	plugin       *nfsMountPlugin
+}
+
+func (nfs *nfsMount) SetUp() error {
+	flags := uintptr(0)
+	if strings.Contains(nfs.mountOptions, "ro") {
+		flags = mount.FlagReadOnly
+	}
+	exportDir := nfs.exportPath
+	mountDir := nfs.GetPath()
+	exists, err := exists(mountDir)
+	if err != nil {
+		glog.Errorf("Mount directory is invalid: %v, %v", mountDir, err)
+	}
+	if !exists {
+		os.MkdirAll(mountDir, 0750)
+	}
+	// NFS Mount format is server:/export/path /mount
+	syscall.Mount(nfs.server+":"+exportDir, mountDir, "nfs", mount.FlagBind|flags, "")
+	if err != nil {
+		glog.Errorf("exportDir is: %v", exportDir)
+		glog.Errorf("mountDir is: %v", mountDir)
+		glog.Errorf("server is: %v", nfs.server)
+		glog.Errorf("NFS mounting failed: %v", err)
+		return err
+	}
+
+	return nil
+}
+
+func exists(path string) (bool, error) {
+	_, err := os.Stat(path)
+	if err == nil {
+		return true, nil
+	}
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	return false, err
+}
+
+func (nfs *nfsMount) GetPath() string {
+	name := nfsMountPluginName
+	return nfs.plugin.host.GetPodVolumeDir(nfs.podUID, volume.EscapePluginName(name), nfs.volName)
+}
+
+func (nfs *nfsMount) GetServer() string {
+	return nfs.server
+}
+
+func (nfs *nfsMount) GetMountOptions() string {
+	return nfs.mountOptions
+}
+
+// TearDown does nothing.
```

Comment:
Yeah, this one is an old comment. The implementation of TearDown cleans  up the dir.